{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "lrtRbP3fAvj8"
      },
      "outputs": [],
      "source": [
        "pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "12b61013-4fdd-4a86-a992-321754786b77"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...\n",
            "📦 Installing...\n",
            "📌 Adjusting configuration...\n",
            "🩹 Patching environment...\n",
            "⏲ Done in 0:00:38\n",
            "🔁 Restarting kernel...\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XnbIt77YBOT9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2afdc996-04da-417d-9374-1de2713cc4ff"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - seqtk\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
            "    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge\n",
            "    conda-4.11.0               |   py37h89c1867_1         983 KB  conda-forge\n",
            "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         1.3 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_1\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.11.0-py37h89c1867_1\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00, 11.04it/s]\n",
            "seqtk-1.3            | 38 KB     | : 100% 1.0/1 [00:00<00:00, 15.65it/s]\n",
            "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00, 17.54it/s]\n",
            "conda-4.11.0         | 983 KB    | : 100% 1.0/1 [00:00<00:00,  2.93it/s]\n",
            "Preparing transaction: / \b\bdone\n",
            "Verifying transaction: \\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "7ff3da03-faea-4b70-dfdb-c2a2e7fce865"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        97.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00,  8.91it/s]\n",
            "openjdk-8.0.312      | 97.6 MB   | : 100% 1.0/1 [00:17<00:00, 17.24s/it]               \n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 30.74it/s]\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\bdone\n",
            "Executing transaction: - \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "5ca83228-582f-4ffa-c613-48b6c1dcb8eb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 2.1 MB 14.0 MB/s \n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 5.5 MB 29.4 MB/s \n",
            "\u001b[?25hCollecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[K     |████████████████████████████████| 12.5 MB 35.6 MB/s \n",
            "\u001b[?25hCollecting matplotlib\n",
            "  Downloading matplotlib-3.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.2 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.2 MB 48.2 MB/s \n",
            "\u001b[?25hCollecting pandas\n",
            "  Downloading pandas-1.3.5-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 11.3 MB 46.2 MB/s \n",
            "\u001b[?25hCollecting numpy\n",
            "  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)\n",
            "Collecting scipy\n",
            "  Downloading scipy-1.7.3-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 38.1 MB 1.2 MB/s \n",
            "\u001b[?25hCollecting seaborn\n",
            "  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)\n",
            "\u001b[K     |████████████████████████████████| 292 kB 70.8 MB/s \n",
            "\u001b[?25hCollecting packaging>=20.0\n",
            "  Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "\u001b[K     |████████████████████████████████| 40 kB 5.3 MB/s \n",
            "\u001b[?25hCollecting cycler>=0.10\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Collecting pyparsing>=2.2.1\n",
            "  Downloading pyparsing-3.0.7-py3-none-any.whl (98 kB)\n",
            "\u001b[K     |████████████████████████████████| 98 kB 8.2 MB/s \n",
            "\u001b[?25hCollecting python-dateutil>=2.7\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[K     |████████████████████████████████| 247 kB 72.3 MB/s \n",
            "\u001b[?25hCollecting fonttools>=4.22.0\n",
            "  Downloading fonttools-4.29.1-py3-none-any.whl (895 kB)\n",
            "\u001b[K     |████████████████████████████████| 895 kB 64.5 MB/s \n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1\n",
            "  Downloading kiwisolver-1.3.2-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 66.9 MB/s \n",
            "\u001b[?25hCollecting pillow>=6.2.0\n",
            "  Downloading Pillow-9.0.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.3 MB 37.7 MB/s \n",
            "\u001b[?25hRequirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib->intervene) (1.15.0)\n",
            "Collecting pytz>=2017.3\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "\u001b[K     |████████████████████████████████| 503 kB 49.6 MB/s \n",
            "\u001b[?25hCollecting pysam\n",
            "  Downloading pysam-0.18.0-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 14.9 MB 54.5 MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: intervene, macs2, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546118 sha256=f68860d2310d384f7b5907edd4ac7594f6865b87cba2eb7544d31688e48654a0\n",
            "  Stored in directory: /root/.cache/pip/wheels/f3/d2/c1/18fa6122c36fdc731af4f16c59c356606b12a890a578294044\n",
            "  Building wheel for macs2 (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=MACS2-2.2.7.1-cp37-cp37m-linux_x86_64.whl size=1790967 sha256=5d6a751acb0b7a4c7e0c659e0bffd234455136777e8b06bc7a5602d022481dc3\n",
            "  Stored in directory: /root/.cache/pip/wheels/4e/34/10/f6222945f7c5b091e34b058a01a842ba9b1d960f2ab0c8e16f\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp37-cp37m-linux_x86_64.whl size=13654258 sha256=37dbc3cb0148f749b7ed6fa6c41ad72a1451743189259bf20d85b82a313ba1e1\n",
            "  Stored in directory: /root/.cache/pip/wheels/7a/44/0d/3a7449885adaf8ebb157da8c3c834a712f48b3b3b84ba51dda\n",
            "Successfully built intervene macs2 pybedtools\n",
            "Installing collected packages: pyparsing, pytz, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, scipy, pysam, pandas, matplotlib, seaborn, pybedtools, macs2, intervene\n",
            "Successfully installed cycler-0.11.0 fonttools-4.29.1 intervene-0.6.5 kiwisolver-1.3.2 macs2-2.2.7.1 matplotlib-3.5.1 numpy-1.21.5 packaging-21.3 pandas-1.3.5 pillow-9.0.1 pybedtools-0.9.0 pyparsing-3.0.7 pysam-0.18.0 python-dateutil-2.8.2 pytz-2021.3 scipy-1.7.3 seaborn-0.11.2\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numpy",
                  "pyparsing"
                ]
              }
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install macs2 intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "1e08b885-1f7f-4f37-d903-4f8ad174ddd0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 0s (6,377 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155332 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "5a093fa2-3030-400f-d758-b72e3e2de8f2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings\n",
            "Suggested packages:\n",
            "  bowtie2-examples python-cryptography-doc python-cryptography-vectors\n",
            "  python-enum34-doc python-genshi python-lxml python-lockfile-doc\n",
            "  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc\n",
            "  python-rdflib-doc python-rdflib-tools python-socks python-ntlm\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc\n",
            "  python-asn1crypto python-avro python-cachecontrol python-certifi\n",
            "  python-cffi-backend python-chardet python-cryptography python-enum34\n",
            "  python-html5lib python-idna python-ipaddress python-isodate python-lockfile\n",
            "  python-mistune python-openssl python-pkg-resources python-pyparsing\n",
            "  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml\n",
            "  python-schema-salad python-shellescape python-six python-sparqlwrapper\n",
            "  python-typing python-urllib3 python-webencodings samtools\n",
            "0 upgraded, 36 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 9,743 kB of archives.\n",
            "After this operation, 46.1 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pkg-resources all 39.0.1-2 [128 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-six all 1.11.0-2 [11.3 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-isodate all 0.6.0-1 [23.8 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-pyparsing all 2.2.0+dfsg1-2 [52.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib all 4.2.1-2 [237 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-certifi all 2018.1.18-2 [144 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-chardet all 3.0.4-1 [80.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-idna all 2.6-1 [32.4 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-urllib3 all 1.22-1ubuntu0.18.04.2 [86.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-requests all 2.18.4-2ubuntu0.1 [58.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-ruamel.yaml amd64 0.15.34-1 [183 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-cachecontrol all 0.11.7-1 [15.2 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-lockfile all 1:0.12.2-2 [14.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-mistune all 0.8.3-2 [13.8 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-rdflib-jsonld all 0.4.0-3 [12.2 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-typing all 3.6.2-1 [22.2 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-avro all 1.8.2+dfsg-1 [23.2 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libjs-bootstrap all 3.3.7+dfsg-2 [228 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-schema-salad all 2.6.20171201034858-3 [260 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-shellescape all 3.4.1-2 [4,432 B]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu bionic/universe amd64 cwltool all 1.0.20180302231433-1 [143 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libhts2 amd64 1.7-2 [300 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs-doc all 8.10.0~dfsg-2ubuntu0.4 [752 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-asn1crypto all 0.24.0-1 [72.7 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-cffi-backend amd64 1.11.5-1 [63.4 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-enum34 all 1.1.6-2 [34.8 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-ipaddress all 1.0.17-1 [18.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python-cryptography amd64 2.1.4-1ubuntu1.4 [276 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-webencodings all 0.5-2 [10.3 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-html5lib all 0.999999999-1 [83.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu bionic/main amd64 python-openssl all 17.5.0-1ubuntu1 [41.3 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu bionic/universe amd64 python-sparqlwrapper all 1.7.6-2 [21.8 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu bionic/universe amd64 samtools amd64 1.7-1 [288 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bowtie2 amd64 2.3.4.1-1 [1,174 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libc-ares2 amd64 1.14.0-1ubuntu0.1 [37.5 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 nodejs amd64 8.10.0~dfsg-2ubuntu0.4 [4,796 kB]\n",
            "Fetched 9,743 kB in 0s (35.0 MB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python-pkg-resources.\n",
            "(Reading database ... 155383 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python-pkg-resources_39.0.1-2_all.deb ...\n",
            "Unpacking python-pkg-resources (39.0.1-2) ...\n",
            "Selecting previously unselected package python-six.\n",
            "Preparing to unpack .../01-python-six_1.11.0-2_all.deb ...\n",
            "Unpacking python-six (1.11.0-2) ...\n",
            "Selecting previously unselected package python-isodate.\n",
            "Preparing to unpack .../02-python-isodate_0.6.0-1_all.deb ...\n",
            "Unpacking python-isodate (0.6.0-1) ...\n",
            "Selecting previously unselected package python-pyparsing.\n",
            "Preparing to unpack .../03-python-pyparsing_2.2.0+dfsg1-2_all.deb ...\n",
            "Unpacking python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Selecting previously unselected package python-rdflib.\n",
            "Preparing to unpack .../04-python-rdflib_4.2.1-2_all.deb ...\n",
            "Unpacking python-rdflib (4.2.1-2) ...\n",
            "Selecting previously unselected package python-certifi.\n",
            "Preparing to unpack .../05-python-certifi_2018.1.18-2_all.deb ...\n",
            "Unpacking python-certifi (2018.1.18-2) ...\n",
            "Selecting previously unselected package python-chardet.\n",
            "Preparing to unpack .../06-python-chardet_3.0.4-1_all.deb ...\n",
            "Unpacking python-chardet (3.0.4-1) ...\n",
            "Selecting previously unselected package python-idna.\n",
            "Preparing to unpack .../07-python-idna_2.6-1_all.deb ...\n",
            "Unpacking python-idna (2.6-1) ...\n",
            "Selecting previously unselected package python-urllib3.\n",
            "Preparing to unpack .../08-python-urllib3_1.22-1ubuntu0.18.04.2_all.deb ...\n",
            "Unpacking python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Selecting previously unselected package python-requests.\n",
            "Preparing to unpack .../09-python-requests_2.18.4-2ubuntu0.1_all.deb ...\n",
            "Unpacking python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Selecting previously unselected package python-ruamel.yaml.\n",
            "Preparing to unpack .../10-python-ruamel.yaml_0.15.34-1_amd64.deb ...\n",
            "Unpacking python-ruamel.yaml (0.15.34-1) ...\n",
            "Selecting previously unselected package python-cachecontrol.\n",
            "Preparing to unpack .../11-python-cachecontrol_0.11.7-1_all.deb ...\n",
            "Unpacking python-cachecontrol (0.11.7-1) ...\n",
            "Selecting previously unselected package python-lockfile.\n",
            "Preparing to unpack .../12-python-lockfile_1%3a0.12.2-2_all.deb ...\n",
            "Unpacking python-lockfile (1:0.12.2-2) ...\n",
            "Selecting previously unselected package python-mistune.\n",
            "Preparing to unpack .../13-python-mistune_0.8.3-2_all.deb ...\n",
            "Unpacking python-mistune (0.8.3-2) ...\n",
            "Selecting previously unselected package python-rdflib-jsonld.\n",
            "Preparing to unpack .../14-python-rdflib-jsonld_0.4.0-3_all.deb ...\n",
            "Unpacking python-rdflib-jsonld (0.4.0-3) ...\n",
            "Selecting previously unselected package python-typing.\n",
            "Preparing to unpack .../15-python-typing_3.6.2-1_all.deb ...\n",
            "Unpacking python-typing (3.6.2-1) ...\n",
            "Selecting previously unselected package python-avro.\n",
            "Preparing to unpack .../16-python-avro_1.8.2+dfsg-1_all.deb ...\n",
            "Unpacking python-avro (1.8.2+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../17-libjs-bootstrap_3.3.7+dfsg-2_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Selecting previously unselected package python-schema-salad.\n",
            "Preparing to unpack .../18-python-schema-salad_2.6.20171201034858-3_all.deb ...\n",
            "Unpacking python-schema-salad (2.6.20171201034858-3) ...\n",
            "Selecting previously unselected package python-shellescape.\n",
            "Preparing to unpack .../19-python-shellescape_3.4.1-2_all.deb ...\n",
            "Unpacking python-shellescape (3.4.1-2) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../20-cwltool_1.0.20180302231433-1_all.deb ...\n",
            "Unpacking cwltool (1.0.20180302231433-1) ...\n",
            "Selecting previously unselected package libhts2:amd64.\n",
            "Preparing to unpack .../21-libhts2_1.7-2_amd64.deb ...\n",
            "Unpacking libhts2:amd64 (1.7-2) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../22-nodejs-doc_8.10.0~dfsg-2ubuntu0.4_all.deb ...\n",
            "Unpacking nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Selecting previously unselected package python-asn1crypto.\n",
            "Preparing to unpack .../23-python-asn1crypto_0.24.0-1_all.deb ...\n",
            "Unpacking python-asn1crypto (0.24.0-1) ...\n",
            "Selecting previously unselected package python-cffi-backend.\n",
            "Preparing to unpack .../24-python-cffi-backend_1.11.5-1_amd64.deb ...\n",
            "Unpacking python-cffi-backend (1.11.5-1) ...\n",
            "Selecting previously unselected package python-enum34.\n",
            "Preparing to unpack .../25-python-enum34_1.1.6-2_all.deb ...\n",
            "Unpacking python-enum34 (1.1.6-2) ...\n",
            "Selecting previously unselected package python-ipaddress.\n",
            "Preparing to unpack .../26-python-ipaddress_1.0.17-1_all.deb ...\n",
            "Unpacking python-ipaddress (1.0.17-1) ...\n",
            "Selecting previously unselected package python-cryptography.\n",
            "Preparing to unpack .../27-python-cryptography_2.1.4-1ubuntu1.4_amd64.deb ...\n",
            "Unpacking python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Selecting previously unselected package python-webencodings.\n",
            "Preparing to unpack .../28-python-webencodings_0.5-2_all.deb ...\n",
            "Unpacking python-webencodings (0.5-2) ...\n",
            "Selecting previously unselected package python-html5lib.\n",
            "Preparing to unpack .../29-python-html5lib_0.999999999-1_all.deb ...\n",
            "Unpacking python-html5lib (0.999999999-1) ...\n",
            "Selecting previously unselected package python-openssl.\n",
            "Preparing to unpack .../30-python-openssl_17.5.0-1ubuntu1_all.deb ...\n",
            "Unpacking python-openssl (17.5.0-1ubuntu1) ...\n",
            "Selecting previously unselected package python-sparqlwrapper.\n",
            "Preparing to unpack .../31-python-sparqlwrapper_1.7.6-2_all.deb ...\n",
            "Unpacking python-sparqlwrapper (1.7.6-2) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../32-samtools_1.7-1_amd64.deb ...\n",
            "Unpacking samtools (1.7-1) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../33-bowtie2_2.3.4.1-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.4.1-1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../34-libc-ares2_1.14.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../35-nodejs_8.10.0~dfsg-2ubuntu0.4_amd64.deb ...\n",
            "Unpacking nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-idna (2.6-1) ...\n",
            "Setting up nodejs-doc (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "Setting up python-asn1crypto (0.24.0-1) ...\n",
            "Setting up python-mistune (0.8.3-2) ...\n",
            "Setting up python-certifi (2018.1.18-2) ...\n",
            "Setting up python-avro (1.8.2+dfsg-1) ...\n",
            "Setting up python-pkg-resources (39.0.1-2) ...\n",
            "Setting up bowtie2 (2.3.4.1-1) ...\n",
            "Setting up python-ruamel.yaml (0.15.34-1) ...\n",
            "Setting up python-cffi-backend (1.11.5-1) ...\n",
            "Setting up python-six (1.11.0-2) ...\n",
            "Setting up python-isodate (0.6.0-1) ...\n",
            "Setting up python-shellescape (3.4.1-2) ...\n",
            "Setting up python-enum34 (1.1.6-2) ...\n",
            "Setting up libjs-bootstrap (3.3.7+dfsg-2) ...\n",
            "Setting up python-lockfile (1:0.12.2-2) ...\n",
            "Setting up python-ipaddress (1.0.17-1) ...\n",
            "Setting up libc-ares2:amd64 (1.14.0-1ubuntu0.1) ...\n",
            "Setting up libhts2:amd64 (1.7-2) ...\n",
            "Setting up python-webencodings (0.5-2) ...\n",
            "Setting up python-typing (3.6.2-1) ...\n",
            "Setting up python-pyparsing (2.2.0+dfsg1-2) ...\n",
            "Setting up python-urllib3 (1.22-1ubuntu0.18.04.2) ...\n",
            "Setting up python-chardet (3.0.4-1) ...\n",
            "Setting up nodejs (8.10.0~dfsg-2ubuntu0.4) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python-rdflib (4.2.1-2) ...\n",
            "Setting up python-sparqlwrapper (1.7.6-2) ...\n",
            "Setting up python-rdflib-jsonld (0.4.0-3) ...\n",
            "Setting up python-cryptography (2.1.4-1ubuntu1.4) ...\n",
            "Setting up python-requests (2.18.4-2ubuntu0.1) ...\n",
            "Setting up python-cachecontrol (0.11.7-1) ...\n",
            "Setting up python-schema-salad (2.6.20171201034858-3) ...\n",
            "Setting up python-html5lib (0.999999999-1) ...\n",
            "Setting up samtools (1.7-1) ...\n",
            "Setting up python-openssl (17.5.0-1ubuntu1) ...\n",
            "Setting up cwltool (1.0.20180302231433-1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.3) ...\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjawt.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libjli.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: file /usr/local/lib/libnet.so is truncated\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/python3.7/dist-packages/ideep4py/lib/libmkldnn.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "ba12adda-f230-49d6-c562-0aa102611030"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda\n",
            "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
            "    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        27.9 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
            "  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.82it/s]\n",
            "bowtie2-2.3.5.1      | 12.0 MB   | : 100% 1.0/1 [00:02<00:00,  2.37s/it]               \n",
            "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.55s/it]               \n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "62l1Q2ymbENA",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ce16d0b8-e14f-4dfb-a40f-0df8c1acbb5c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"1.8.0_312\"\n",
            "OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)\n",
            "OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "8bpgwFgcbgRd",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8dee8047-d482-40ef-dc12-2a806bdde7cd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 16:51:23--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  20.5MB/s    in 0.5s    \n",
            "\n",
            "2022-03-09 16:51:23 (20.5 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "yTcTjOTJblcm",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "daf7374f-53a3-41c7-f941-24610c1d9716"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "PsLuZDt4b7jT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "845862f1-8efa-4453-a6b7-30572f161cb4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "lFWnMg0uU0sJ",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cfc1fa32-cfee-4481-ec15-d69ec0f37bda"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 16:51:24--  https://www.encodeproject.org/files/ENCFF834DXB/@@download/ENCFF834DXB.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/22/816ea1ba-a5d5-40b8-ac34-2751530e6581/ENCFF834DXB.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF834DXB.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=gtp3oG%2FsfayFj%2BRDqIZqYT4JncA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974285 [following]\n",
            "--2022-03-09 16:51:25--  https://encode-public.s3.amazonaws.com/2016/10/22/816ea1ba-a5d5-40b8-ac34-2751530e6581/ENCFF834DXB.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF834DXB.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=gtp3oG%2FsfayFj%2BRDqIZqYT4JncA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974285\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.221.203\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.221.203|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 786938500 (750M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF834DXB.fastq.gz’\n",
            "\n",
            "ENCFF834DXB.fastq.g 100%[===================>] 750.48M  18.5MB/s    in 42s     \n",
            "\n",
            "2022-03-09 16:52:08 (17.8 MB/s) - ‘ENCFF834DXB.fastq.gz’ saved [786938500/786938500]\n",
            "\n",
            "--2022-03-09 16:52:08--  https://www.encodeproject.org/files/ENCFF983KWQ/@@download/ENCFF983KWQ.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/20/81f6e136-d418-4604-b0c3-26c0f82c8444/ENCFF983KWQ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF983KWQ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=M4yINzJE0VwJ5E4hkGemUBKPjnA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974328 [following]\n",
            "--2022-03-09 16:52:08--  https://encode-public.s3.amazonaws.com/2016/10/20/81f6e136-d418-4604-b0c3-26c0f82c8444/ENCFF983KWQ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF983KWQ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=M4yINzJE0VwJ5E4hkGemUBKPjnA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974328\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.205.99\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.205.99|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1705692380 (1.6G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF983KWQ.fastq.gz’\n",
            "\n",
            "ENCFF983KWQ.fastq.g 100%[===================>]   1.59G  18.5MB/s    in 90s     \n",
            "\n",
            "2022-03-09 16:53:39 (18.1 MB/s) - ‘ENCFF983KWQ.fastq.gz’ saved [1705692380/1705692380]\n",
            "\n",
            "--2022-03-09 16:53:39--  https://www.encodeproject.org/files/ENCFF902MPL/@@download/ENCFF902MPL.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/21/7d9d9398-d65b-4d4d-ab63-11a0b36dbd0b/ENCFF902MPL.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF902MPL.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=MwamtGB8AYRvJt6BJmW10dR%2Fom8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974420 [following]\n",
            "--2022-03-09 16:53:40--  https://encode-public.s3.amazonaws.com/2016/10/21/7d9d9398-d65b-4d4d-ab63-11a0b36dbd0b/ENCFF902MPL.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF902MPL.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=MwamtGB8AYRvJt6BJmW10dR%2Fom8%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oaVqQMnEmneqkFtRKPZDDnIJ4bNQydY0niVwSZk0bAxg3QP2MaoNIDo52gagzV%2FfCNSmbXw6LRcA4IoLZhWeaUdHO3iBiYYRwyCvuTpxa553mPndRAJonqL7vZOcqysKWHtdP7p%2B%2FJX%2FPOVA9qbKakK%2F7QaM5LZ9GQT6LEatskY3cjuRwmUmPCKPWhlxt2bRE6zao2T%2F1iJJdBJfIjZQLCWLuFoiltCw1Gkb5r9JB7ID9DKJtTr1vn6F3EOf98RmBXAkkKIL1P%2Fokg05Tq4JXkVB%2BGvhhoYSJAKrsNC0U3p%2FnKBVyOeVpJVaI9QkF1d%2BEqN4n8ax1Oobdy06rlR2iJWHpTgwqbtukjgkONS9yK8kjZ7lTWknne2rcEnmMo4c3SuRaz%2FEG1t7AOfKKxLl9ekOzLRp1N9aMpe0l1f9m9nqRtnePkfN1AYb3xIA5ysGex9YBRNCzs8RfzlVRR2hfsZtn%2F45Ozc3MY0SFhMCF5OwjDxq6ORBjqlAaGux7xl8dtP7XLEZjavixTz0lnCDGB6fPZbnusSxM6GSo%2FmqCkYC63T7z9%2BgbbVNXAS32wADaK48EP4lfL3QpzHEkqmsqYfSDKo5vt90t47hjcXfAtl%2FbuonIWDVg%2BaAdRzGiw4M7e9Ox%2F4pCMQg0Yaph%2FEe3ux%2B2ImIAWNm7OLXg5Z6%2BgpEynnVx0hwTlxphOWeGw8qYId7x2pGXd60QuSb%2FSmCw%3D%3D&Expires=1646974420\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.217.59\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.217.59|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1679358434 (1.6G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF902MPL.fastq.gz’\n",
            "\n",
            "ENCFF902MPL.fastq.g 100%[===================>]   1.56G  18.3MB/s    in 90s     \n",
            "\n",
            "2022-03-09 16:55:10 (17.9 MB/s) - ‘ENCFF902MPL.fastq.gz’ saved [1679358434/1679358434]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF834DXB/@@download/ENCFF834DXB.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF983KWQ/@@download/ENCFF983KWQ.fastq.gz\n",
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF902MPL/@@download/ENCFF902MPL.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d /content/ENCFF834DXB.fastq.gz /content/ENCFF983KWQ.fastq.gz /content/ENCFF902MPL.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC (или multiQC может быть использован в работе)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5XM6jBcuGjMV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "3abfab5c-57ac-48c7-a0f0-71cc8f45164a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF834DXB.fastq\n",
            "Approx 5% complete for ENCFF834DXB.fastq\n",
            "Approx 10% complete for ENCFF834DXB.fastq\n",
            "Approx 15% complete for ENCFF834DXB.fastq\n",
            "Approx 20% complete for ENCFF834DXB.fastq\n",
            "Approx 25% complete for ENCFF834DXB.fastq\n",
            "Approx 30% complete for ENCFF834DXB.fastq\n",
            "Approx 35% complete for ENCFF834DXB.fastq\n",
            "Approx 40% complete for ENCFF834DXB.fastq\n",
            "Approx 45% complete for ENCFF834DXB.fastq\n",
            "Approx 50% complete for ENCFF834DXB.fastq\n",
            "Approx 55% complete for ENCFF834DXB.fastq\n",
            "Approx 60% complete for ENCFF834DXB.fastq\n",
            "Approx 65% complete for ENCFF834DXB.fastq\n",
            "Approx 70% complete for ENCFF834DXB.fastq\n",
            "Approx 75% complete for ENCFF834DXB.fastq\n",
            "Approx 80% complete for ENCFF834DXB.fastq\n",
            "Approx 85% complete for ENCFF834DXB.fastq\n",
            "Approx 90% complete for ENCFF834DXB.fastq\n",
            "Approx 95% complete for ENCFF834DXB.fastq\n",
            "Analysis complete for ENCFF834DXB.fastq\n",
            "Started analysis of ENCFF983KWQ.fastq\n",
            "Approx 5% complete for ENCFF983KWQ.fastq\n",
            "Approx 10% complete for ENCFF983KWQ.fastq\n",
            "Approx 15% complete for ENCFF983KWQ.fastq\n",
            "Approx 20% complete for ENCFF983KWQ.fastq\n",
            "Approx 25% complete for ENCFF983KWQ.fastq\n",
            "Approx 30% complete for ENCFF983KWQ.fastq\n",
            "Approx 35% complete for ENCFF983KWQ.fastq\n",
            "Approx 40% complete for ENCFF983KWQ.fastq\n",
            "Approx 45% complete for ENCFF983KWQ.fastq\n",
            "Approx 50% complete for ENCFF983KWQ.fastq\n",
            "Approx 55% complete for ENCFF983KWQ.fastq\n",
            "Approx 60% complete for ENCFF983KWQ.fastq\n",
            "Approx 65% complete for ENCFF983KWQ.fastq\n",
            "Approx 70% complete for ENCFF983KWQ.fastq\n",
            "Approx 75% complete for ENCFF983KWQ.fastq\n",
            "Approx 80% complete for ENCFF983KWQ.fastq\n",
            "Approx 85% complete for ENCFF983KWQ.fastq\n",
            "Approx 90% complete for ENCFF983KWQ.fastq\n",
            "Approx 95% complete for ENCFF983KWQ.fastq\n",
            "Analysis complete for ENCFF983KWQ.fastq\n",
            "Started analysis of ENCFF902MPL.fastq\n",
            "Approx 5% complete for ENCFF902MPL.fastq\n",
            "Approx 10% complete for ENCFF902MPL.fastq\n",
            "Approx 15% complete for ENCFF902MPL.fastq\n",
            "Approx 20% complete for ENCFF902MPL.fastq\n",
            "Approx 25% complete for ENCFF902MPL.fastq\n",
            "Approx 30% complete for ENCFF902MPL.fastq\n",
            "Approx 35% complete for ENCFF902MPL.fastq\n",
            "Approx 40% complete for ENCFF902MPL.fastq\n",
            "Approx 45% complete for ENCFF902MPL.fastq\n",
            "Approx 50% complete for ENCFF902MPL.fastq\n",
            "Approx 55% complete for ENCFF902MPL.fastq\n",
            "Approx 60% complete for ENCFF902MPL.fastq\n",
            "Approx 65% complete for ENCFF902MPL.fastq\n",
            "Approx 70% complete for ENCFF902MPL.fastq\n",
            "Approx 75% complete for ENCFF902MPL.fastq\n",
            "Approx 80% complete for ENCFF902MPL.fastq\n",
            "Approx 85% complete for ENCFF902MPL.fastq\n",
            "Approx 90% complete for ENCFF902MPL.fastq\n",
            "Approx 95% complete for ENCFF902MPL.fastq\n",
            "Analysis complete for ENCFF902MPL.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF834DXB.fastq\n",
        "!./FastQC/fastqc /content/ENCFF983KWQ.fastq\n",
        "!./FastQC/fastqc /content/ENCFF902MPL.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk."
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Видим из выдачи fastqc, что качество первой реплики и контроля не самое лучшее - постараемся подрезать чтения, чтобы его улучшить"
      ],
      "metadata": {
        "id": "gMXO40rMe4i0"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Qellza2PI7fT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "57252fad-af63-4500-d56d-751677facc15"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF834DXB.fastq ENCFF834DXB_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 18361776 Surviving: 17283292 (94.13%) Dropped: 1078484 (5.87%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ],
      "source": [
        "!trimmomatic SE -phred33 ENCFF834DXB.fastq ENCFF834DXB_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "PWBBF50gI7mI",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0c473ef1-43d5-433a-f2ef-f5297aa8d8fe"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF834DXB_trimmed.fastq\n",
            "Approx 5% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 10% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 15% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 20% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 25% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 30% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 35% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 40% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 45% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 50% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 55% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 60% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 65% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 70% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 75% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 80% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 85% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 90% complete for ENCFF834DXB_trimmed.fastq\n",
            "Approx 95% complete for ENCFF834DXB_trimmed.fastq\n",
            "Analysis complete for ENCFF834DXB_trimmed.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc ENCFF834DXB_trimmed.fastq"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF902MPL.fastq ENCFF902MPL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FFtF3Y-rpa_r",
        "outputId": "df9a2f74-dfc6-4da6-bf7e-49c33fc0a0fe"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF902MPL.fastq ENCFF902MPL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.io.FileInputStream.open(FileInputStream.java:195)\n",
            "\tat java.io.FileInputStream.<init>(FileInputStream.java:138)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 38682658 Surviving: 37572200 (97.13%) Dropped: 1110458 (2.87%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc ENCFF902MPL_trimmed.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6jn1HQm4pb0c",
        "outputId": "23efedb6-5a9e-4c13-a1d0-9e41059e83b5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF902MPL_trimmed.fastq\n",
            "Approx 5% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 10% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 15% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 20% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 25% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 30% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 35% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 40% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 45% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 50% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 55% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 60% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 65% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 70% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 75% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 80% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 85% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 90% complete for ENCFF902MPL_trimmed.fastq\n",
            "Approx 95% complete for ENCFF902MPL_trimmed.fastq\n",
            "Analysis complete for ENCFF902MPL_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "После подрезания чтений качество немного улучшилось\n"
      ],
      "metadata": {
        "id": "lwUc8AD2rAhV"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n",
        "\n",
        "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
        "\n",
        "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Была выбрана 21-ая хромосома\n"
      ],
      "metadata": {
        "id": "14GDG3j3quDs"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "N4JIGZOsGiY2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f1c3b184-4678-4e92-ea38-1d9727cd9c6d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 17:13:23--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr21.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 12709705 (12M) [application/x-gzip]\n",
            "Saving to: ‘chr21.fa.gz’\n",
            "\n",
            "chr21.fa.gz         100%[===================>]  12.12M  7.18MB/s    in 1.7s    \n",
            "\n",
            "2022-03-09 17:13:26 (7.18 MB/s) - ‘chr21.fa.gz’ saved [12709705/12709705]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr21.fa.gz\n",
        "!gzip -d chr21.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "bhraQ6NdGiPN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a1ff6031-076f-4a75-f85f-744ee386f7ee"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr21.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "bmax according to bmaxDivN setting: 10022154\n",
            "Using parameters --bmax 7516616 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 7516616 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:01\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 4.00886e+07 (target: 7516615)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 40088619 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:21\n",
            "Returning block of 40088620 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 11820664\n",
            "fchr[G]: 20005908\n",
            "fchr[T]: 28232289\n",
            "fchr[$]: 40088619\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 17557951 bytes to primary EBWT file: chromosome_index.1.bt2\n",
            "Wrote 10022160 bytes to secondary EBWT file: chromosome_index.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 40088619\n",
            "    bwtLen: 40088620\n",
            "    sz: 10022155\n",
            "    bwtSz: 10022155\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 2505539\n",
            "    offsSz: 10022156\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 208795\n",
            "    numLines: 208795\n",
            "    ebwtTotLen: 13362880\n",
            "    ebwtTotSz: 13362880\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:00:29\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:00\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:00\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 10022154\n",
            "Using parameters --bmax 7516616 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 7516616 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:01\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 4.00886e+07 (target: 7516615)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 1\n",
            "  No samples; assembling all-inclusive block\n",
            "  Sorting block of length 40088619 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:20\n",
            "Returning block of 40088620 for bucket 1\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 11820664\n",
            "fchr[G]: 20005908\n",
            "fchr[T]: 28232289\n",
            "fchr[$]: 40088619\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 17557951 bytes to primary EBWT file: chromosome_index.rev.1.bt2\n",
            "Wrote 10022160 bytes to secondary EBWT file: chromosome_index.rev.2.bt2\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 40088619\n",
            "    bwtLen: 40088620\n",
            "    sz: 10022155\n",
            "    bwtSz: 10022155\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 2505539\n",
            "    offsSz: 10022156\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 208795\n",
            "    numLines: 208795\n",
            "    ebwtTotLen: 13362880\n",
            "    ebwtTotSz: 13362880\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:00:28\n",
            "CPU times: user 492 ms, sys: 65.8 ms, total: 557 ms\n",
            "Wall time: 57.7 s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr21.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f139e06d-f6f2-4667-b59e-dee146c80b02"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "17283292 reads; of these:\n",
            "  17283292 (100.00%) were unpaired; of these:\n",
            "    15393136 (89.06%) aligned 0 times\n",
            "    417150 (2.41%) aligned exactly 1 time\n",
            "    1473006 (8.52%) aligned >1 times\n",
            "10.94% overall alignment rate\n",
            "CPU times: user 2.47 s, sys: 318 ms, total: 2.79 s\n",
            "Wall time: 5min 24s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF834DXB_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_DXB.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2e9180af-bde2-4883-a2ce-9488835a7713"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "40166682 reads; of these:\n",
            "  40166682 (100.00%) were unpaired; of these:\n",
            "    35422229 (88.19%) aligned 0 times\n",
            "    1085078 (2.70%) aligned exactly 1 time\n",
            "    3659375 (9.11%) aligned >1 times\n",
            "11.81% overall alignment rate\n",
            "CPU times: user 5.84 s, sys: 738 ms, total: 6.58 s\n",
            "Wall time: 12min 57s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF983KWQ.fastq \\\n",
        "          -S bowtie2_res/res_KWQ.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF902MPL_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_MPL.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5ecaspblStZ5",
        "outputId": "f44e219a-33c8-4440-d951-77418c72fd2f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "37572200 reads; of these:\n",
            "  37572200 (100.00%) were unpaired; of these:\n",
            "    31572119 (84.03%) aligned 0 times\n",
            "    1187983 (3.16%) aligned exactly 1 time\n",
            "    4812098 (12.81%) aligned >1 times\n",
            "15.97% overall alignment rate\n",
            "CPU times: user 6.34 s, sys: 807 ms, total: 7.14 s\n",
            "Wall time: 14min 16s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "#!samtools view -h -S -b -o bowtie2_res/res_DXB.bam bowtie2_res/res_DXB.sam\n",
        "#!samtools view -h -S -b -o bowtie2_res/res_KWQ.bam bowtie2_res/res_KWQ.sam\n",
        "#!samtools view -h -S -b -o bowtie2_res/res_MPL.bam bowtie2_res/res_MPL.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uX1ps4VDp_Ae",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "66c95498-b190-4915-97cc-75f7837466cb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "mkdir: cannot create directory ‘macs2_DXB’: File exists\n",
            "mkdir: cannot create directory ‘macs2_KWQ’: File exists\n"
          ]
        }
      ],
      "source": [
        "!mkdir macs2_DXB\n",
        "!mkdir macs2_KWQ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a30d52d3-b8b8-499c-d23c-f7a628bac4c5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 09 Mar 2022 18:12:53: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_DXB.sam -c bowtie2_res/res_MPL.sam -f SAM --outdir macs2_DXB\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_DXB.sam']\n",
            "# control file = ['bowtie2_res/res_MPL.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 09 Mar 2022 18:12:53: #1 read tag files... \n",
            "INFO  @ Wed, 09 Mar 2022 18:12:53: #1 read treatment tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:09: #1.2 read input tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:18:  9000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:33:  24000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:35:  26000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:40:  31000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 tag size is determined as 36 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 tag size = 36.0 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  total tags in treatment: 1890156 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  tags after filtering in treatment: 1503830 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  Redundant rate of treatment: 0.20 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  total tags in control: 6000081 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  tags after filtering in control: 3771157 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1  Redundant rate of control: 0.37 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #1 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 Build Peak Model... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 number of paired peaks: 6608 \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: start model_add_line... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: start X-correlation... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: end of X-cor \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 predicted fragment length is 35 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2 alternative fragment length(s) may be 35,108,149,178,257,294,387,491 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #2.2 Generate R script for model : macs2_DXB/NA_model.r \n",
            "WARNING @ Wed, 09 Mar 2022 18:13:49: #2 Since the d (35) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 09 Mar 2022 18:13:49: #2 You may need to consider one of the other alternative d(s): 35,108,149,178,257,294,387,491 \n",
            "WARNING @ Wed, 09 Mar 2022 18:13:49: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #3 Call peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 09 Mar 2022 18:13:49: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:03: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: #4 Write output xls file... macs2_DXB/NA_peaks.xls \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: #4 Write broad peak in broadPeak format file... macs2_DXB/NA_peaks.broadPeak \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: #4 Write broad peak in bed12/gappedPeak format file... macs2_DXB/NA_peaks.gappedPeak \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_DXB.sam \\\n",
        "    -c bowtie2_res/res_MPL.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2_DXB"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_KWQ.sam \\\n",
        "    -c bowtie2_res/res_MPL.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2_KWQ"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "br0pCaVUURW7",
        "outputId": "141c136b-a35f-41b6-83aa-b5b12e30d1fa"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 09 Mar 2022 18:14:12: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_KWQ.sam -c bowtie2_res/res_MPL.sam -f SAM --outdir macs2_KWQ\n",
            "# ARGUMENTS LIST:\n",
            "# name = NA\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_KWQ.sam']\n",
            "# control file = ['bowtie2_res/res_MPL.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: #1 read tag files... \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:12: #1 read treatment tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:29:  19000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:36:  27000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:40:  31000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:42:  33000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:43:  34000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:45:  36000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:49: #1.2 read input tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:14:58:  9000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:13:  24000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:15:  26000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:20:  31000000 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 tag size is determined as 36 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 tag size = 36.0 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  total tags in treatment: 4744453 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  tags after filtering in treatment: 3151673 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  Redundant rate of treatment: 0.34 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  total tags in control: 6000081 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  tags after filtering in control: 3771157 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1  Redundant rate of control: 0.37 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #1 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #2 Build Peak Model... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:28: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #2 number of paired peaks: 13934 \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: start model_add_line... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: start X-correlation... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: end of X-cor \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #2 finished! \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #2 predicted fragment length is 35 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #2 alternative fragment length(s) may be 35,151,192,242,286,351,377 bps \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #2.2 Generate R script for model : macs2_KWQ/NA_model.r \n",
            "WARNING @ Wed, 09 Mar 2022 18:15:29: #2 Since the d (35) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 09 Mar 2022 18:15:29: #2 You may need to consider one of the other alternative d(s): 35,151,192,242,286,351,377 \n",
            "WARNING @ Wed, 09 Mar 2022 18:15:29: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #3 Call peaks... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:29: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:46: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:56: #4 Write output xls file... macs2_KWQ/NA_peaks.xls \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:56: #4 Write broad peak in broadPeak format file... macs2_KWQ/NA_peaks.broadPeak \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:56: #4 Write broad peak in bed12/gappedPeak format file... macs2_KWQ/NA_peaks.gappedPeak \n",
            "INFO  @ Wed, 09 Mar 2022 18:15:56: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f1c9bceb-d22d-4008-f158-fa9f83a81cb7"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-03-09 17:48:50--  https://www.encodeproject.org/files/ENCFF090XFB/@@download/ENCFF090XFB.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/01/31/2854b499-9022-425b-b163-7866f22da6d2/ENCFF090XFB.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF090XFB.bed.gz&AWSAccessKeyId=ASIATGZNGCNX4XLA2UVH&Signature=ZEmcubUvFC7F2RXBNXz8U7oZE3o%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHIaCXVzLXdlc3QtMiJGMEQCIGGuz9teVbSH5o%2FgzS7vd5fFAjycUzc%2BhYmH3p2fxva7AiBaiMZJMAYDXKNpW%2FSoqOXNLjSuGzL8THCiFoHPSPXr5iqDBAjb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMMHG8R%2BNTw5JRlMImKtcDl50sw80Bcn2tK3kCMHiUob2VZrW28GkIC7PQIDcPkg2NihtwqOCgNyv7GZk3Cqa3cdgnpZAIknSqUuHfW8bBYGgkMIIjUQsBaNMKX4C39y3x1rrRhhNlNsVocpsy4aTIZeiS6hGPEvUPgv8pG3%2BQOCEew0ZyWjzTiuWm5PSRFW9jogkJhCFqspMpHn6DJ1I%2BATokIE7ntDuiyAfJ80xD4o9n%2FHM%2BQxhOSJgukuxjAJxrkLp2FWjgOd4NszSv42nmAl6FUaTT2jmsY5F17DustDDxln5mqqk5niMWnw2gzbmAEa9jhI6YuMHD%2FPTeOwJRNxmdk7Ysw3feGdrHMa4et7JAGk6EUL61Is4h%2Fux8qbAuGbfpSbeA7NpFCy%2Bx70snFQIlmHbFHriz7TUCrw39GRrEzEFnGCsCsbtSOvl9Jqf8yVh%2BM0O0JAqAH%2BsUR3ucAsANkXm0qgs6zg1JnN1gwGev%2BgPpFm7IiRvnqzO7C5KoBUZFhnABUW3o87hn1J91oP%2BauqL%2FMCh0qjMPBQjrtjA6adlldwar13AbsQSFiBswx48BbYaewH4NT36kAejlkCPdQb0wEzPHbY66OeJnjMVZRUmWjTO3lU94H9Fvbl4msP7xjtuoMOHHo5EGOqYB50ONA1ezOwiBUpqxL7LppE8AAqIoSYSlKYqCVPTSQWIBI%2FEyiIPvXJw4oxTvwp4CbPQqUv%2F%2BkbrbBSOGVQhheJSUaF%2BvC0qO8DibNM%2FjXiAaJLHkvoTAGGDBQpxRH1cuJEDnvw58%2FzJcUOSc5NwxPzTxNLYEhJQqCdTgfu5bFHhf2I8Itt%2B0kXIIDImRECIIOQpQDY%2B%2BAL0Up%2BvjGs6FxNp6ZzvDEA%3D%3D&Expires=1646977730 [following]\n",
            "--2022-03-09 17:48:50--  https://encode-public.s3.amazonaws.com/2017/01/31/2854b499-9022-425b-b163-7866f22da6d2/ENCFF090XFB.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF090XFB.bed.gz&AWSAccessKeyId=ASIATGZNGCNX4XLA2UVH&Signature=ZEmcubUvFC7F2RXBNXz8U7oZE3o%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHIaCXVzLXdlc3QtMiJGMEQCIGGuz9teVbSH5o%2FgzS7vd5fFAjycUzc%2BhYmH3p2fxva7AiBaiMZJMAYDXKNpW%2FSoqOXNLjSuGzL8THCiFoHPSPXr5iqDBAjb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMMHG8R%2BNTw5JRlMImKtcDl50sw80Bcn2tK3kCMHiUob2VZrW28GkIC7PQIDcPkg2NihtwqOCgNyv7GZk3Cqa3cdgnpZAIknSqUuHfW8bBYGgkMIIjUQsBaNMKX4C39y3x1rrRhhNlNsVocpsy4aTIZeiS6hGPEvUPgv8pG3%2BQOCEew0ZyWjzTiuWm5PSRFW9jogkJhCFqspMpHn6DJ1I%2BATokIE7ntDuiyAfJ80xD4o9n%2FHM%2BQxhOSJgukuxjAJxrkLp2FWjgOd4NszSv42nmAl6FUaTT2jmsY5F17DustDDxln5mqqk5niMWnw2gzbmAEa9jhI6YuMHD%2FPTeOwJRNxmdk7Ysw3feGdrHMa4et7JAGk6EUL61Is4h%2Fux8qbAuGbfpSbeA7NpFCy%2Bx70snFQIlmHbFHriz7TUCrw39GRrEzEFnGCsCsbtSOvl9Jqf8yVh%2BM0O0JAqAH%2BsUR3ucAsANkXm0qgs6zg1JnN1gwGev%2BgPpFm7IiRvnqzO7C5KoBUZFhnABUW3o87hn1J91oP%2BauqL%2FMCh0qjMPBQjrtjA6adlldwar13AbsQSFiBswx48BbYaewH4NT36kAejlkCPdQb0wEzPHbY66OeJnjMVZRUmWjTO3lU94H9Fvbl4msP7xjtuoMOHHo5EGOqYB50ONA1ezOwiBUpqxL7LppE8AAqIoSYSlKYqCVPTSQWIBI%2FEyiIPvXJw4oxTvwp4CbPQqUv%2F%2BkbrbBSOGVQhheJSUaF%2BvC0qO8DibNM%2FjXiAaJLHkvoTAGGDBQpxRH1cuJEDnvw58%2FzJcUOSc5NwxPzTxNLYEhJQqCdTgfu5bFHhf2I8Itt%2B0kXIIDImRECIIOQpQDY%2B%2BAL0Up%2BvjGs6FxNp6ZzvDEA%3D%3D&Expires=1646977730\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.216.114\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.216.114|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1086944 (1.0M) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF090XFB.bed.gz’\n",
            "\n",
            "ENCFF090XFB.bed.gz  100%[===================>]   1.04M  1.08MB/s    in 1.0s    \n",
            "\n",
            "2022-03-09 17:48:52 (1.08 MB/s) - ‘ENCFF090XFB.bed.gz’ saved [1086944/1086944]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF090XFB/@@download/ENCFF090XFB.bed.gz\n",
        "!gzip -d ENCFF090XFB.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "84fe91cd-b5b7-4a12-ad67-ee3c0ce2a0da"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1_DXB.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2_DXB.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2_DXB/NA_peaks.broadPeak ENCFF090XFB.bed --filenames --output venn_results/venn1_DXB.jpg\n",
        "!intervene venn -i ENCFF090XFB.bed macs2_DXB/NA_peaks.broadPeak --filenames --output venn_results/venn2_DXB.jpg"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "7qqn3n2cRdkt",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "20d9d72f-8bd6-4196-de95-95dd7c5005f2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn1_KWQ.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n",
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn_results/venn2_KWQ.jpg. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2_KWQ/NA_peaks.broadPeak ENCFF090XFB.bed --filenames --output venn_results/venn1_KWQ.jpg\n",
        "!intervene venn -i ENCFF090XFB.bed macs2_KWQ/NA_peaks.broadPeak --filenames --output venn_results/venn2_KWQ.jpg"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "name": "Дз-2 ChIP-Seq Григорьев Final.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}